conans/client/tools/intel.py

```diff
@@ -0,0 +1,107 @@
+        elif arch == "x86_64":
+            intel_arch = "EM64T"
+        else:
+            raise ConanException("don't know how to find Intel compiler on %s" % arch)
+        if is_win64():
+            base = r"SOFTWARE\WOW6432Node\Intel\Suites\{version}".format(version=version)
+        else:
+            base = r"SOFTWARE\Intel\Suites\{version}".format(version=version)
+        from six.moves import winreg  # @UnresolvedImport
+        path = base + r"\Defaults\C++\{arch}".format(arch=intel_arch)
+        subkey = _system_registry_key(winreg.HKEY_LOCAL_MACHINE, path, "SubKey")
+        if not subkey:
+            raise ConanException("unable to find Intel C++ compiler installation")
+        path = base + r"\{subkey}\C++".format(subkey=subkey)
+        installation_path = _system_registry_key(winreg.HKEY_LOCAL_MACHINE, path, "LatestDir")
+        if not installation_path:
+            raise ConanException("unable to find Intel C++ compiler installation")
+    else:
+        raise ConanException("don't know how to find Intel compiler on %s" % system)
+    return installation_path
+
+
+def compilervars_command(settings, arch=None, compiler_version=None, force=False):
+    """
+    https://software.intel.com/en-us/intel-system-studio-cplusplus-compiler-user-and-reference-guide-using-compilervars-file
+    :return:
+    """
+    if "PSTLROOT" in os.environ and not force:
+        return "echo Conan:compilervars already set"
+    compiler_version = compiler_version or settings.get_safe("compiler.version")
+    arch = arch or settings.get_safe("arch")
+    system = platform.system()
+    cvars = "compilervars.bat" if system == "Window" else "compilervars.sh"
+    command = os.path.join(intel_installation_path(version=compiler_version, arch=arch), "bin", cvars)
+    if system != "Windows":
+        command = ". " + command  # dot is more portable than source
+    if arch == "x86_64":
+        command += " -arch intel64"
+        if system != "Windows":
+            command = "COMPILERVARS_ARCHITECTURE=intel64 " + command
+    elif arch == "x86":
+        command += " -arch ia32"
+        if system != "Windows":
+            command = "COMPILERVARS_ARCHITECTURE=ia32 " + command
+    else:
+        raise ConanException("don't know how to call %s for %s" % (cvars, arch))
+    if system == "Darwin":
+        command += " -platform mac"
+        command = "COMPILERVARS_PLATFORM=mac " + command
+    elif system == "Linux":
+        command += " -platform linux"
+        command = "COMPILERVARS_PLATFORM=linux " + command
+    elif system == "Windows":
+        command += " -platform windows"
+    else:
+        raise ConanException("don't know how to call %s for %s" % (cvars, system))
+    compiler_base = settings.get_safe("compiler.base")
+    if compiler_base == "Visual Studio":
+        base_version = settings.get_safe("compiler.base.version")
+        if base_version:
+            command += "vs%s" % MSVS_YEAR.get(base_version)
```

Comment:
Needs to be prefixed with a space. Also, according to the help of compilervars.bat, the visual studio version must come before the platform (again I've never actually used platform).

Reply:
added missing space. as there is no platform anymore, I think it's good now